### ISSUE TITLE
Some C++ code highlighting issues

### ISSUE THREAD

**Issue by yzxoi** [cpeditor/cpeditor#641]
**Describe the bug**

In CP Editor, I used the Dracula highlight theme, but found that it was a little different from the normal theme. For example, in Sublime Text, single quote characters will be highlighted in yellow, but in CP Editor, single quote characters will not be highlighted.

Not only does this problem occur with Dracula highlighting themes, but other themes also appear.

**To Reproduce**

Write C++ programs and use highlight themes

**Screenshots**

![image.png](https://i.loli.net/2020/10/28/9UaLV1QTj6d4AfK.png)

**Environment**

 - OS: Windows 10
 - CP Editor Version: 6.7.1

**Additional context**

**Comment by Ravenclaw-OIer** [comment]
Please wait, it should be fixed if we move to KSyntaxHighlighting, the current QCodeEditor is not elegant in dealing with this.

**Comment by yzxoi** [comment]
> Please wait, it should be fixed if we move to KSyntaxHighlighting (#516), the current QCodeEditor is not elegant in dealing with this.

Okay, so when will it move to KSyntaxHighlighting?The current editor is really uncomfortable.

**Comment by coder3101** [comment]
> Okay, so when will it move to KSyntaxHighlighting?The current editor is really uncomfortable.

Syntax highlighting is a hard task, currently, the Code Editor uses `RegEx` for matching keywords, numbers, functions to highlight them. But obviously regex is not enough, we need a parser based highlighter. KSH(KSyntax Highlighting) is parser-based that we have chosen. It is by KDE team. We could do that move to KSH in a month but we want to delay it because it is a big change, possibly requiring change to 90% of cpeditor source files. Plus these changes are breaking, like current code editor themes will not work with KSH. It needs much testing, validation before we can put it to public use.

It is so big of a change that it will close 5-6 issues directly. Provide code-folding, auto-complete, custom themes, and so much more. 

So, while it may be uncomfortable for you to use current editor, believe me it will be even more uncomfortable for you to use a half baked, untested code editor. If you still want to try out the KSH, you can [build that branch](https://github.com/cpeditor/cpeditor/tree/ksyntax-highlighting) from source. or [use one of build artifacts](https://github.com/cpeditor/cpeditor/suites/1320979160/artifacts/21025955) with `KSyntaxHighlighting.dll`.

**Comment by coder3101** [comment]
Closed as duplicate of #174
